src/components/checkbox/checkbox.jsx

```diff
@@ -0,0 +1,132 @@
+import React, {
+  forwardRef,
+  useEffect,
+  useImperativeHandle,
+  useLayoutEffect,
+  useRef,
+  useState,
+} from 'react';
+import PropTypes from 'prop-types';
+import FormControl from '../form-control/form-control.jsx';
+import useMounted from '../../hooks/use-mounted.js';
+import Icon from '../icon/icon.jsx';
+import styles from './checkbox.css';
+import cautionSvg from '../../svgs/caution.svg';
+
+const Checkbox = forwardRef(function Checkbox(props, forwardedRef) {
+  const fallbackRef = useRef();
+  const ref = forwardedRef || fallbackRef;
+  const inputRef = useRef();
+  const mounted = useMounted();
+  const [validationMessage, setValidationMessage] = useState(props.validationMessage);
+
+  useEffect(() => {
+    if (!mounted.current) return;
+
+    inputRef.current.checked = props.checked;
+  }, [props.checked]);
+
+  useEffect(() => {
+    setValidationMessage(props.validationMessage);
+  }, [props.validationMessage]);
+
+  useLayoutEffect(() => {
+    inputRef.current.setCustomValidity(validationMessage);
+  }, [validationMessage]);
+
+  useImperativeHandle(ref, () => ({
+    id: props.id,
+    name: props.name,
+    get value() {
+      if (props.value) return props.value;
+
+      return inputRef.current.checked ? 'on' : 'off';
+    },
+    get checked() {
+      return inputRef.current.checked;
+    },
+    get dirty() {
+      const providedValue = props.checked || false;
+      return providedValue !== this.checked;
+    },
+  }));
+
+  const className = props.className || styles.input;
```

Comment:
Please use default props instead of the logical assignment.

Reply:
d33d055d784c4fd2e19e67819f23de989b8bc394